python-avd/pyavd/_eos_designs/shared_utils/router_general.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+from pyavd._utils import get
+
+if TYPE_CHECKING:
+    from . import SharedUtils
+
+
+class RouterGeneralMixin:
+    """
+    Mixin Class providing a subset of SharedUtils.
+
+    Class should only be used as Mixin to the SharedUtils class.
+    Using type-hint on self to get proper type-hints on attributes across all Mixins.
+    """
+
+    @cached_property
+    def router_general(self: SharedUtils) -> dict | None:
+        """Router general set based on "router_general_settings" data-model."""
+        router_general = get(self.hostvars, "router_general_settings")
+        if not router_general:
+            return None
+
+        return router_general
```

Comment:
This is the same as just returning directly since `get()` returns None if required is not True.